clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -489,13 +490,33 @@ static Value *EmitISOVolatileStore(CodeGenFunction &CGF, const CallExpr *E) {
   return Store;
 }
 
+static CallInst *CreateBuiltinCallWithAttr(CodeGenFunction &CGF,
+                                           llvm::Function *FPBuiltinF,
+                                           ArrayRef<Value *> Args,
+                                           unsigned ID) {
+  llvm::CallInst *CI = CGF.Builder.CreateCall(FPBuiltinF, Args);
+  unsigned BuiltinID = CGF.getCurrentBuiltinID();
+  StringRef Name = CGF.CGM.getContext().BuiltinInfo.getName(BuiltinID);
+  llvm::AttributeList AttrList;
+  CGF.CGM.getFPAccuracyFuncAttributes(Name, AttrList, ID,
+                                      FPBuiltinF->getReturnType());
+  CI->setAttributes(AttrList);
+  return CI;
+}
+
 // Emit a simple mangled intrinsic that has 1 argument and a return type
 // matching the argument type. Depending on mode, this may be a constrained
 // floating-point intrinsic.
 static Value *emitUnaryMaybeConstrainedFPBuiltin(CodeGenFunction &CGF,
                                 const CallExpr *E, unsigned IntrinsicID,
                                 unsigned ConstrainedIntrinsicID) {
   llvm::Value *Src0 = CGF.EmitScalarExpr(E->getArg(0));
+  if (CGF.CGM.getCodeGenOpts().FPAccuracy) {    
```

Comment:
Can you do something here to see if there is an accuracy requirement for this function? It's better to not use the fpbuiltin intrinsic if it isn't needed?

Reply:
Can you please explain why? I will add a comment.